core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1890,47 +1890,66 @@ class SparkContext(config: SparkConf) extends Logging {
             throw new IllegalArgumentException(
               s"Directory ${path} is not allowed for addJar")
           }
-          path
+          Seq(path)
         } catch {
           case NonFatal(e) =>
             logError(s"Failed to add $path to Spark environment", e)
-            null
+            Nil
         }
       } else {
-        path
+        Seq(path)
       }
     }
 
     if (path == null || path.isEmpty) {
       logWarning("null or empty path specified as parameter to addJar")
     } else {
-      val key = if (path.contains("\\") && Utils.isWindows) {
+      var schema = ""
```

Comment:
`val schema = uri.getScheme`? Please avoid to use `var` where possible.

Reply:
Should consider about window path,  how about current change?